From 4b34c626bb1fb58baedca9d05f961f5123c396fe Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:00:18 -0400
Subject: [fs] btrfs: fix snprintf usage by send's gen_unique_name

Message-id: <1407179109-13714-132-git-send-email-sandeen@redhat.com>
Patchwork-id: 87021
O-Subject: [RHEL7.1 PATCH 131/422] Btrfs: fix snprintf usage by send's gen_unique_name
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Filipe David Borba Manana <fdmanana@gmail.com>

commit f74b86d85533a98ef7f573487af38f9dd514becb
Author: Filipe David Borba Manana <fdmanana@gmail.com>
Date:   Tue Jan 21 23:36:38 2014 +0000

    Btrfs: fix snprintf usage by send's gen_unique_name

    The buffer size argument passed to snprintf must account for the
    trailing null byte added by snprintf, and it returns a value >= then
    sizeof(buffer) when the string can't fit in the buffer.

    Since our buffer has a size of 64 characters, and the maximum orphan
    name we can generate is 63 characters wide, we must pass 64 as the
    buffer size to snprintf, and not 63.

    Signed-off-by: Filipe David Borba Manana <fdmanana@gmail.com>
    Reviewed-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/send.c b/fs/btrfs/send.c
index aa60cbe..fc1f0ab 100644
--- a/fs/btrfs/send.c
+++ b/fs/btrfs/send.c
@@ -1336,7 +1336,7 @@ static int gen_unique_name(struct send_ctx *sctx,
   return -ENOMEM;
 
  while (1) {
-  len = snprintf(tmp, sizeof(tmp) - 1, "o%llu-%llu-%llu",
+  len = snprintf(tmp, sizeof(tmp), "o%llu-%llu-%llu",
     ino, gen, idx);
   if (len >= sizeof(tmp)) {
    /* should really not happen */
-- 
1.7.1